src/components/input/input.js

```diff
@@ -684,7 +685,7 @@ function mdMaxlengthDirective($animate, $mdUtil) {
 
       // Force the value into a string since it may be a number,
       // which does not have a length property.
-      charCountEl.text(String(element.val() || value || '').length + ' / ' + maxlength);
+      charCountEl.text(String(element.val().trim || value || '').length + ' / ' + maxlength);
```

Comment:
Instead of always trimming the text we should make this more dependent to the model

* Similar to the `ng-maxlength` directive: [See source](https://github.com/angular/angular.js/blob/master/src/ng/directive/validators.js#L266)

This would mean that the counter respects the `ng-trim` attribute.

Reply:
I looked at the `ng-maxlength` directive linked here, but I don't really see it interacting with the `ng-trim` attribute in any way.